src/App.tsx

```diff
@@ -1,10 +1,59 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Error } from './components/Error/Error';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Todolist } from './components/Todolist/Todolist';
+import { ErrorMessage } from './types/ErrorMessage';
+import { TodoStatus } from './types/TodoStatus';
 
-const USER_ID = 0;
+const USER_ID = 10890;
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.ALL);
+  const [isError, setIsError] = useState<ErrorMessage>(ErrorMessage.NOERROR);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setVisibleTodos)
+      .catch(() => {
+        setIsError(ErrorMessage.LOADERROR);
+      });
+  }, []);
+
+  const handleCloseError = () => {
+    setIsError(ErrorMessage.NOERROR);
+  };
+
+  const filterTodos = useCallback((todoStatus: string) => {
+    if (todoStatus === TodoStatus.COMPLETED) {
+      return visibleTodos.filter(todo => todo.completed);
+    }
+
+    if (todoStatus === TodoStatus.ACTIVE) {
+      return visibleTodos.filter(todo => !todo.completed);
+    }
+
+    return visibleTodos;
+  }, [status, visibleTodos]);
+
+  const filteredTodos = filterTodos(status);
```

Comment:
We don't use the function anywhere else; can't we calculate `filteredTodos` here?